### ISSUE TITLE
YAML-CPP

### ISSUE THREAD

**Issue by amzeratul** [conan-io/wishlist#12]
[YAML-CPP](https://github.com/jbeder/yaml-cpp/) - A library for parsing YAML 1.2 in C++.

Uses CMake. Works on VC++, GCC and Clang. Readme.md suggests that it uses Boost, but I don't see anything on CMakeLists.txt.

**Comment by theirix** [comment]
I'll try to package a new version. Current stable version 0.5.3 is the last version that requires Boost. In newer versions yaml-cpp will switch to C++11 and drop boost dependency. It makes sense to wait for a new version and save efforts on packaging.

**Comment by IntegralLee** [comment]
I put together a yaml-cpp 0.5.3 package at [yaml-cpp/0.5.3@moonforged/stable](https://conan.io/source/yaml-cpp/0.5.3/moonforged/stable).

**Comment by danimtb** [comment]
There is ongoing work by bincrafters (@theirix): https://github.com/bincrafters/conan-yaml-cpp

Is this ready for conan-center?

cc/ @solvingj @uilianries

**Comment by uilianries** [comment]
Hi! @danimtb 

Our recipe is working good. I'm using at my job.

I just requested now the inclusion on Conan center.

We could close this issue after to have it on Conan center. WDYT?

**Comment by danimtb** [comment]
Perfect, will review it asap. Many thanks!

**Comment by danimtb** [comment]
I see the inclusion request is already there but I realized it has dependencies to bincrafters' modular boost 😅

**Comment by uilianries** [comment]
I could change to use Conan community flavor boost/1.67@conan/stable

**Comment by danimtb** [comment]
Let's do that to have it approved! Thanks

**Comment by uilianries** [comment]
I created some PRs to improve our current yaml-cpp recipes:

https://github.com/bincrafters/conan-yaml-cpp/pulls

I'll merge ASAP

**Comment by uilianries** [comment]
All yaml-cpp branches are updated! :smile_cat:

**Comment by uilianries** [comment]
For which version? yaml-cpp 0.6.x uses C++14 so msvc 12 is not able to
build.

**Comment by danimtb** [comment]
Yes, you are right. Then raise errors that invalid configuration:

```
def configure(self):
    if self.settings.compiler == "Visual Studio" and self.settings.compiler.version == "12":
        raise Exception("Visual Studio 12 not supported: Library needs C++14 standard")
```

**Comment by uilianries** [comment]
Absolutely!
